flink-runtime/src/main/java/org/apache/flink/runtime/security/modules/HadoopModule.java

```diff
@@ -75,6 +83,10 @@ public void install() throws SecurityInstallException {
                                         new File(fileLocation), hadoopConfiguration);
                         loginUser.addCredentials(credentials);
                     }
+                    tgtRenewalExecutorService =
```

Comment:
Can we renew the tokens if the login happened from the `HADOOP_TOKEN_FILE_LOCATION`?

Reply:
Flink is never renew tokens but re-obtains them. The DT framework overwrites tokens coming from other sources other then providers.

Reply:
Wait a minute here, just realized the place where this comment added. Maybe you mix up TGT and delegation token. This is TGT renewal but HADOOP_TOKEN_FILE_LOCATION provides delegation tokens in a file. TGT can come from 2 sources:
* Get from keytab -> this we're renewing
* TGT cache file (which is basically `/tmp/krb...{uid}` or something) -> this case it's the user's responsibility to keep it alive (most probably w/ re-occuring `kinit` cmd)

Reply:
I see. Thanks for the explanation!